extensions/basic_auth/plugin.cc

```diff
@@ -0,0 +1,287 @@
+                    "Request denied by Basic Auth check. Invalid "
+                    "username and/or password",
+                    "", {});
+}
+}  // namespace
+
+bool PluginRootContext::onConfigure(size_t size) {
+  // Parse configuration JSON string.
+  if (size > 0 && !configure(size)) {
+    LOG_WARN("configuration has errors initialization will not continue.");
+    return false;
+  }
+  return true;
+}
+
+bool PluginRootContext::configure(size_t configuration_size) {
+  auto configuration_data = getBufferBytes(WasmBufferType::PluginConfiguration,
+                                           0, configuration_size);
+  // Parse configuration JSON string.
+  auto result = ::Wasm::Common::JsonParse(configuration_data->view());
+  if (!result.has_value()) {
+    LOG_WARN(absl::StrCat("cannot parse plugin configuration JSON string: ",
+                          configuration_data->view()));
+    return false;
+  }
+  // j is a JsonObject holds configuration data
+  auto j = result.value();
+  // basic_auth_configuration_ container has the following example structure
+  //{
+  // "GET":{
+  //    { "/products",
+  //      "prefix",
+  //      ["YWRtaW46YWRtaW4="]
+  //    },
+  //    { "/api/products/reviews",
+  //      "exact",
+  //      ["YWRtaW46YWRtaW4=", "ARtaW46YWRW4="]
+  //    }
+  // },
+  // "POST":{
+  //     { "/wiki",
+  //      "prefix",
+  //      ["YWRtaW46YWRtaW4=", "AWRtaW46YWRtaW4="]
+  //    }
+  //  },
+  // "DELETE":{
+  //    { "/api/store/product/id/two",
+  //      "exact",
+  //      ["AWRtaW46YWRtaW4="]
+  //    }
+  //  }
+  //}
+  if (!JsonArrayIterate(
+          j, "basic_auth_rules", [&](const json& configuration) -> bool {
+            std::string match;
+            std::string request_path;
+            std::vector<std::string> request_methods;
+            if (!JsonObjectIterate(
+                    configuration, "request_path",
+                    [&](std::string pattern) -> bool {
+                      match = pattern;
+                      auto path = JsonGetField<std::string>(
+                                      configuration["request_path"], pattern)
+                                      .value_or("");
+                      auto path_string = JsonValueAs<std::string>(path);
+                      if (path_string.second !=
+                          Wasm::Common::JsonParserResultDetail::OK) {
+                        LOG_WARN("unexpected request path");
+                        return false;
+                      }
+                      request_path = path;
+                      return true;
+                    })) {
+              LOG_WARN("Failed to parse configuration for request path.");
+              return false;
+            }
+            if (request_path == "" || match == "") {
+              return false;
+            }
+            if (!JsonArrayIterate(
+                    configuration, "request_methods",
+                    [&](const json& methods) -> bool {
```

Comment:
should not be plural. Should be method and next line should use method_string variable.